creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1004,6 +927,103 @@ class SnapbackSM {
+       * Enqueue SyncRequests against healthy secondaries, and enqueue UpdateReplicaSetOps against unhealthy secondaries
+       */
+      let replicaSetNodesToObserve = [
+        { endpoint: secondary1, spId: secondary1SpID },
+        { endpoint: secondary2, spId: secondary2SpID }
+      ]
+
+      if (primary === this.endpoint) {
+        // filter out false-y values to account for incomplete replica sets
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => entry.endpoint && entry.spId)
+
+        /**
+         * If either secondary is in `unhealthyPeers` list, add it to `unhealthyReplicas` list
+         */
+        const userSecondarySyncMetrics = await this._computeUserSecondarySyncSuccessRates(nodeUser, [secondary1, secondary2])
+        for (const secondary of replicaSetNodesToObserve) {
+          const secUserSyncSuccessRate = userSecondarySyncMetrics[secondary.endpoint]['SuccessRate']
+          if (secUserSyncSuccessRate < this.MinimumSecondaryUserSyncSuccessPercent ||
+            unhealthyPeers.has(secondary.endpoint) ||
+            this.peerSetManager.endpointToSPIdMap[secondary.endpoint] !== secondary.spId) {
+            unhealthyReplicas.push(secondary.endpoint)
+          } else {
+            potentialSyncRequests.push({ ...nodeUser, endpoint: secondary.endpoint })
+          }
+        }
+
+        /**
+         * If any unhealthy replicas found for user, enqueue an updateReplicaSetOp for later processing
+         */
+        if (unhealthyReplicas.length > 0) {
+          requiredUpdateReplicaSetOps.push({ ...nodeUser, unhealthyReplicas })
+        }
+
+        /**
+         * If this node is secondary for user, check both secondaries for health and enqueue SyncRequests against healthy secondaries
+         * Ignore unhealthy secondaries for now
+         */
+      } else {
+        // filter out false-y values to account for incomplete replica sets and filter out the
+        // the self node
+        replicaSetNodesToObserve = [{ endpoint: primary, spId: primarySpID }, ...replicaSetNodesToObserve]
+        replicaSetNodesToObserve = replicaSetNodesToObserve.filter(entry => {
+          return entry.endpoint && entry.spId && entry.endpoint !== this.endpoint
+        })
+
+        for (const replica of replicaSetNodesToObserve) {
+          // If the map's spId does not match the query's spId, then regardless
+          // of the relationship of the node to the user, issue a reconfig for that node
+          if (this.peerSetManager.endpointToSPIdMap[replica.endpoint] !== replica.spId) {
+            unhealthyReplicas.push(replica.endpoint)
+          } else if (unhealthyPeers.has(replica.endpoint)) {
+            // Else, continue with conducting extra health check if the current observed node is a primary, and
+            // add to `unhealthyReplicas` if observed node is a secondary
+            let addToUnhealthyReplicas = true
```

Comment:
this would be much simpler without the `addToUnhealthyReplicas` var as just
```
if (replica.endpoint === primary && !(await this.peerSetManager.isPrimaryHealthy(primary)) {
  unhealthyRelpicas.push(replica.endpoint)
}
```

Reply:
I like how in my implementation that we can see with the boolean that if snapback encounters unhealthy node, that it will add to unhealthyReplicas. it is only upon encountering a primary and checking again that it is healthy that snapback will not. I think that your suggestion is more concise, but will require some extra logic to handle the case when snapback encounters a secondary. 

im gonna leave it as is unless you feel strongly if that's alright

Reply:
i think its still cleaner with that but yeah nbd